pkg/e2e/operators/cloudingress/deletetapps2.go

```diff
@@ -0,0 +1,146 @@
+			//2.delete the annotation
+			apps2Ingress, _ := getingressController(h, ingressControllerName)
+			log.Printf("The ingresscontroller object annotation currently is: %+v\n", apps2Ingress.ObjectMeta.Annotations)
+			newAnnotations := removeAnnotation(h, ingressControllerName)
+			apps2Ingress = newAnnotations
+			log.Printf("Deleted the Annotation. This ingresscontroller should not belong to CIO anymore. the object's annotation now looks like: %+v\n", apps2Ingress.ObjectMeta.Annotations)
+			//3. Delete secondaryIngress in publishingstrategy
+			removeIngressController(h, ingressControllerName)
+			log.Print("Try to delete the secondaryIngress that doesn't contains the annotation")
+			time.Sleep(time.Duration(360) * time.Second)
+			// check that the ingresscontroller app-e2e-apps was deleted
+			ingressControllerExists(h, ingressControllerName, true)
+			log.Print("Gonna delete the IngressController now")
+			returnAnnotations(h, ingressControllerName)
+			log.Print("Added the Annotations back to IngressOperator")
+			removeIngressController(h, ingressControllerName)
+			log.Print("Called removeIngressController ")
+			ingressControllerExists(h, ingressControllerName, false)
+			log.Print("ingressController Exist shouldn't throw any errors'")
+		})
+	})
+})
+
+func removeIngressController(h *helper.H, name string) {
+	_, exists, index := appIngressExits(h, false, name)
+	// only remove the secondary ingress if it already exist in the publishing strategy
+	if exists {
+		removeAppIngress(h, index)
+		//wait 2 minute for all resources to be deleted
+		time.Sleep(time.Duration(120) * time.Second)
+	}
+}
+func removeAnnotation(h *helper.H, name string) operatorv1.IngressController {
+	var ingressController operatorv1.IngressController
+	log.Print("Gonna start the annotation deletion process")
+	ingresscontroller, err := h.Dynamic().Resource(schema.GroupVersionResource{Group: "operator.openshift.io", Version: "v1", Resource: "ingresscontrollers"}).Namespace("openshift-ingress-operator").Get(context.TODO(), name, metav1.GetOptions{})
+	log.Print("Getting the ingressController object")
+	Expect(err).NotTo(HaveOccurred())
+
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(ingresscontroller.Object, &ingressController)
+	Expect(err).NotTo(HaveOccurred())
+	temp := ingressController.ObjectMeta
+	if temp.Annotations["Owner"] == "cloud-ingress-operator" {
+		delete(temp.Annotations, "Owner")
+	}
+	ingressController.ObjectMeta = temp
+	ingresscontroller.Object, err = runtime.DefaultUnstructuredConverter.ToUnstructured(&ingressController)
+	Expect(err).NotTo(HaveOccurred())
+
+	ingresscontroller, err = h.Dynamic().Resource(schema.GroupVersionResource{Group: "operator.openshift.io", Version: "v1", Resource: "ingresscontrollers"}).Namespace("openshift-ingress-operator").Update(context.TODO(), ingresscontroller, metav1.UpdateOptions{})
+	return ingressController
+}
+func returnAnnotations(h *helper.H, name string) {
```

Comment:
this and the previous function seem to have the same logic. if the purpose is updating the annotation, maybe they could be factorized in a updateAnnotation function, to which we pass the desired annotation directly as a `map[string]string{}`